crates/ra_hir/src/ty.rs

```diff
@@ -413,15 +413,15 @@ impl Ty {
             None => return Ty::Unknown,
             Some(it) => it,
         };
-        let ty = db.type_for_def(typable);
+        let ty = db.type_for_def(typable); // FIXME(csmoe)
```

Comment:
`type_for_def` belongs to `HirDatabase`, how can I fix this without moving this query into `PersistentHirDatabase`?

Reply:
Hm, I think the argument here *should* be `HirDatabase`? Everything type-related should *not* be persistent.

Reply:
the call flow of `type_for_def`:
```
type_for_def <- from_hir_path <- from_hir <- collect_recursive <- impls_in_crate_query
```
then we have a problem, the db from `impls_in_crate_query(db, ...)` belongs to `PersistenHirDB`, then conflicts to `HirDB`.

Reply:
hm, riight, I think we need to move `impls_in_crate_query` to HirDb?